File: Project-Euler/Largest-Prime-Factor/largest_prime_factor.py
'''
Problem Statement: https://www.hackerrank.com/contests/projecteuler/challenges/euler003/problem || https://projecteuler.net/problem=3
Author: striker
'''

def find_largest_prime_factor(n):
    prime_factors = dict()
    while not (n & 1):
        n >>= 1
        if prime_factors.get(2) == None:
            prime_factors[2] = 1
        else:
            prime_factors[2] += 1 
    i = 3
    while i**2 <= n:
        while n % i == 0:
            if prime_factors.get(n) == None:
                prime_factors[i] = 1
            else:
                prime_factors[i] += 1
            n = int(n/i)
        i += 2
    if n > 2:
        if prime_factors.get(n) == None:
            prime_factors[n] = 1
    return max(prime_factors)

def main():
    test = int(input())
    for _ in range(test):
        n = int(input())
        print(find_largest_prime_factor(n))

if __name__ == "__main__":
    main()

File: Project-Euler/Largest-Prime-Factor/largest_prime_factor.c
/*  Problem Statement: https://www.hackerrank.com/contests/projecteuler/challenges/euler003/problem
    Author: striker
*/

#include<stdio.h>
#include<stdlib.h>
#include<inttypes.h>
#include<assert.h>

static uint64_t largest_prime_factor(uint64_t);

int main(void) {
    int8_t test;
    scanf("%"SCNd8, &test);
    assert(test > 0 && test < 11);
    while(test--) {
        uint64_t n;
        scanf("%"SCNu64, &n);
        assert(n > 9 && n < 1000000000001);
        printf("%"PRIu64"\n", largest_prime_factor(n));
    }
    return EXIT_SUCCESS;
}

static uint64_t largest_prime_factor(uint64_t n) {
    uint64_t max_prime_factor = 0;
    if(!(n & 1)) {
        max_prime_factor = 2;
    }
    while(!(n & 1)) {
        n >>= 1;
    }
    for(uint64_t i = 3; (i * i) <= n; i +=2) {
        if(!(n % i)) {
            max_prime_factor = i;
            while(!(n % i)) {
                n /= i;
            }
        }
    }
    if(n > 2) {
        max_prime_factor = n;
    }
    return max_prime_factor;
}